arangod/Aql/AstNode.cpp

```diff
@@ -376,6 +383,57 @@ static bool IsEmptyString(char const* p, size_t length) {
   return true;
 }
 
+/*static*/ std::string AstNode::encodeDataSourceType(
+    char* const name, size_t size, AstNode::DataSourceType type
+) {
+  StringRef suffix;
+
+  switch (type) {
+    case AstNode::DataSourceType::View:
+      suffix = VIEW_NODE_SUFFIX;
+      break;
+    case AstNode::DataSourceType::Collection:
+      suffix = COLLECTION_NODE_SUFFIX;
+      break;
+    default: // shut up compiler
+      break;
+  }
+
+  std::string param(name, size + suffix.size());
+  std::memcpy(&param[0] + size, suffix.data(), suffix.size());
+  return param;
+}
+
+/*static*/ AstNode::DataSourceType AstNode::decodeDataSourceType(
+    std::string& param
+) {
+  auto decodeParam = [](std::string& param, StringRef const& expectedSuffix) -> bool {
+    auto const suffixPos = param.size() - expectedSuffix.size();
```

Comment:
if param.size() is 1 and expectedSuffix is 2 (which it always is), then this will fail badly.

Reply:
fixed